state_machine/docs/state-machine.diagram.md

```diff
@@ -0,0 +1,14 @@
+# Node Finite State Machine
+
+The following diagram displays the various states and events that govern the functionality of the node.
+
+```mermaid
+stateDiagram-v2
+    [*] --> Stopped
+    Consensus_SyncMode --> Consensus_Synced: Consensus_IsCaughtUp
+    Consensus_Unsynched --> Consensus_SyncMode: Consensus_IsSyncing
+    P2P_Bootstrapped --> Consensus_Synced: Consensus_IsCaughtUp
+    P2P_Bootstrapped --> Consensus_Unsynched: Consensus_IsUnsynched
+    P2P_Bootstrapping --> P2P_Bootstrapped: P2P_IsBootstrapped
+    Stopped --> P2P_Bootstrapping: Start
+```
```

Comment:
Answering https://github.com/pokt-network/pocket/pull/520/files#r1109007712

If I recall correctly, the server mode is something that is handled like a parallel process separately.

I think it's out of scope for this PR anyway.